### PR TITLE
FEATURE: hidden site setting to suppress unsecured categories from admins

### PR DESCRIPTION
**Pull request by SamSaffron**
The hidden site setting `suppress_secured_categories_from_admin` will
suppress visibility of categories without explicit access from admins
in a few key areas (category drop downs and topic lists)

It is not intended to be a security wall since admins can amend any site
setting. Instead it is feature that allows hiding the categories from the
UI.

Admins will still be able to see topics in categories without explicit
access using direct URLs or flags.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -3100,4 +3100,22 @@ def reset_last_seen_cache!(user)
...
+  describe '#secure_category_ids' do
+    fab!(:admin) { Fabricate(:admin) }
+    fab!(:group) { Fabricate(:group) }
+    fab!(:private_category) { Fabricate(:private_category, group: group) }
+
+    it 'allows admin to see all secure categories' do
+      expect(admin.secure_category_ids).to include(private_category.id)
+    end
+
+    context 'when SiteSetting.suppress_secured_categories_from_admin' do
```
Comment (tgxworld) [inline spec/models/user_spec.rb:3113]:
```suggestion
    context 'when SiteSetting.suppress_secured_categories_from_admin is true' do
```

File: spec/lib/guardian/topic_guardian_spec.rb [hunk 2]
```diff
@@ -174,4 +176,30 @@
...
+
+    it 'allows admin access to categories without explicit access' do
+      guardian = Guardian.new(admin)
+      list = Topic.where(id: private_topic.id)
+      list = guardian.filter_allowed_categories(list)
+
+      expect(list.count).to eq(1)
+    end
+
+    context 'when SiteSetting.suppress_secured_categories_from_admin is enabled' do
```
Comment (tgxworld) [inline spec/lib/guardian/topic_guardian_spec.rb:190]:
```suggestion
    context 'when SiteSetting.suppress_secured_categories_from_admin is true' do
```

**Review by tgxworld — APPROVED** [review]
Changes look good :+1:

File: config/site_settings.yml [hunk 3]
```diff
@@ -1771,6 +1771,10 @@ security:
     default: false
     client: true
     hidden: true
+  suppress_secured_categories_from_admin:
+    default: false
+    client: true
```
Comment (tgxworld) [inline config/site_settings.yml:1776]: Sorry one more note, does this need to be shipped to the client?

Comment (SamSaffron) [reply config/site_settings.yml:1776]: ohh cut and paste fail ... no not client side thing will fix

**Pull request merged**